src/components/Bridge/BridgeConfirmModal.tsx

```diff
@@ -94,147 +315,74 @@ const BridgeConfirmModal = (props: Props) => {
       new BigNumber(amount).shiftedBy(12).multipliedBy(fees.issueGriefingCollateral),
     );
   }, [amount, fees]);
+  
+  const stepsProps: StepsProps = {
+      amount,
+      assetInfo,
+      bridgeFee,
+      errorMessage,
+      griefingCollateral,
+      isError,
+      isSuccess,
+      selectedAsset,
+      selectedChainFrom,
+      selectedChainTo,
+      theme,
+      txFee,
+      tryAgain,
+      txHash,
+      isPending,
+      onClickConfirmButton,
+      onCloseConfirmModal,
+      setActiveStep 
+    };
+
+  const [steps, setSteps] = useState<StepType[] | null>(null);
+
+  const redeemSteps: StepType[] = useMemo(() => {
+    const redeemStepKeys = Object.keys(RedeemStepsByKeys);
+    return allSteps.filter((step: any) => redeemStepKeys.includes(step.key));
+  }, [allSteps]);
+
+  const issueSteps: StepType[] = useMemo(() => {
+    const StepKeys = Object.keys(IssueStepsByKeys);
+    return allSteps.filter((step: any) => StepKeys.includes(step.key));
+  } , [allSteps]);
+
+  useEffect(() => {
+    if(selectedChainFrom === 'Stellar') {
+      setSteps(issueSteps);
+    } else if (selectedChainFrom === 'Pendulum') {
+      setSteps(redeemSteps);
+    }
+  }, [selectedChainFrom, selectedChainTo]);
```

Comment:
Estas seguro de que issueSteps and redeemSteps no deberían estar en el arreglo de dependencias??